osu.Game.Tournament/IO/TournamentStorage.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Threading;
+using osu.Framework.Logging;
+using osu.Framework.Platform;
+using osu.Game.IO;
+using System.IO;
+using osu.Game.Tournament.Configuration;
+
+namespace osu.Game.Tournament.IO
+{
+    public class TournamentStorage : WrappedStorage
+    {
+        private readonly GameHost host;
+        internal readonly TournamentVideoResourceStore VideoStore;
+        internal readonly Storage ConfigurationStorage;
+        private const string default_tournament = "default";
+        private const string config_directory = "config";
+
+        public TournamentStorage(GameHost host)
+            : base(host.Storage.GetStorageForDirectory("tournaments"), string.Empty)
+        {
+            this.host = host;
+
+            TournamentStorageManager storageConfig = new TournamentStorageManager(host.Storage);
+
+            var currentTournament = storageConfig.Get<string>(StorageConfig.CurrentTournament);
+
+            if (!string.IsNullOrEmpty(currentTournament))
+            {
+                ChangeTargetStorage(UnderlyingStorage.GetStorageForDirectory(currentTournament));
+            }
+            else
+            {
+                Migrate();
+                storageConfig.Set(StorageConfig.CurrentTournament, default_tournament);
+                storageConfig.Save();
+                ChangeTargetStorage(UnderlyingStorage.GetStorageForDirectory(default_tournament));
+            }
+
+            ConfigurationStorage = UnderlyingStorage.GetStorageForDirectory(config_directory);
+
+            VideoStore = new TournamentVideoResourceStore(this);
+            Logger.Log("Using tournament storage: " + GetFullPath(string.Empty));
+        }
+
+        internal void Migrate()
+        {
+            var source = new DirectoryInfo(host.Storage.GetFullPath("tournament"));
+            var destination = new DirectoryInfo(GetFullPath(default_tournament));
+            var cfgDestination = new DirectoryInfo(GetFullPath(default_tournament + Path.DirectorySeparatorChar + config_directory));
+
+            if (!destination.Exists)
+                destination.Create();
+
+            if (!cfgDestination.Exists)
+                destination.CreateSubdirectory(config_directory);
+
+            if (host.Storage.Exists("bracket.json"))
+            {
+                Logger.Log("Migrating bracket to default tournament storage.");
+                var bracketFile = new System.IO.FileInfo(host.Storage.GetFullPath("bracket.json"));
+                moveFile(bracketFile, destination);
+            }
+
+            if (host.Storage.Exists("drawings.txt"))
```

Comment:
why can't you use the existing `Migrate` method? this is mostly copy-paste of existing implementation.